cmd/cfssl/cfssl_test.go

```diff
@@ -1 +1,446 @@
 package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cloudflare/cfssl/api"
+	"github.com/cloudflare/cfssl/auth"
+	"github.com/cloudflare/cfssl/certinfo"
+	"github.com/cloudflare/cfssl/csr"
+	"github.com/cloudflare/cfssl/helpers/testsuite"
+	"github.com/cloudflare/cfssl/info"
+	"github.com/cloudflare/cfssl/scan"
+)
+
+// CFSSL local server to start up
+var server *testsuite.CFSSLServer
+
+// We will test on this address and port.
+const addressToTest = "127.0.0.1"
+const portToTest = 8888
+
+// This is the generic endpoint for our server
+var postStr = fmt.Sprintf("http://%s:%s/api/v1/cfssl/", addressToTest,
+	strconv.Itoa(portToTest))
+
+// JSON requests to ping our server with
+var keyReq, _ = ioutil.ReadFile("testdata/newkeyRequest.json")
+var newcertReq, _ = ioutil.ReadFile("testdata/newcertRequest.json")
+var signReq, _ = ioutil.ReadFile("testdata/signRequest.json")
```

Comment:
Might be worth putting the `ioutil.Readfile` commands in an `init` function. That way you can catch errors so if there is some problem reading the files it will be easier to debug